### PR TITLE
Use new xcomponent reactive api

### PR DESCRIPTION
**Pull request by Julien-Molina**

### CODE REVIEW

File: src/actions/transitionProperties.ts [hunk 1]
```diff
@@ -68,8 +68,8 @@ export const showTransitionProperties = (component: string, stateMachine: string
                         messageType,
                         jsonMessageString,
                         id: firstId,
-                        privateTopic: (privateTopic) ? privateTopic : session.getDefaultPrivateTopic(),
-                        privateTopics: session.getPrivateTopics()
+                        privateTopic: (privateTopic) ? privateTopic : session.privateTopics.getdefaultPublisherTopic(),
```
Comment (maurelio1234) [inline src/actions/transitionProperties.ts]: get*D*efaultPublisherTopic

File: src/actions/transitionProperties.ts [hunk 2]
```diff
@@ -68,8 +68,8 @@ export const showTransitionProperties = (component: string, stateMachine: string
...
+                        privateTopics: session.privateTopics.getSubscriberTopics()
```
Comment (maurelio1234) [inline src/actions/transitionProperties.ts]: why do you have `privateTopic` and `privateTopics` on the same structure?

File: src/utils/sessionXCSpy.ts [hunk 3]
```diff
@@ -1,34 +1,58 @@
...
     }
 
-    init(xcApiFileName: string, serverUrl: string): Promise<Object> {
-        const promiseCreateSession = new Promise((resolve, reject) => {
-            xcomponentapi.createSession(xcApiFileName, serverUrl, (error, session: Session) => {
-                xcomponentapi.setLogLevel(5);
-                if (error) {
+    public init(api: string, serverUrl: string): void {
+        this.promiseCreateSession = new Promise((resolve, reject) => {
+            xcomponentapi.connect(serverUrl).then(connection => {
```
Comment (maurelio1234) [inline src/utils/sessionXCSpy.ts]: that's a perfect use case for async / await :)

File: src/actions/transitionProperties.ts (hunk 2 above)
Comment (Julien-Molina) [reply src/actions/transitionProperties.ts]: A relic from the reactiveApi, will be removed in next PR

File: src/actions/transitionProperties.ts (hunk 1 above)
Comment (Julien-Molina) [reply src/actions/transitionProperties.ts]: The problem is in the API

Comment (maurelio1234) [reply src/actions/transitionProperties.ts]: Ok. created a ticket for that:

https://github.com/xcomponent/ReactiveXComponent.py/issues/32

Comment (Julien-Molina) [reply src/actions/transitionProperties.ts]: Wrong API, you created it in the python API

**Pull request merged**